src/Apps/Auction/Components/RegistrationForm.tsx

```diff
@@ -186,17 +192,53 @@ const validationSchema = Yup.object().shape({
   ),
 })
 
+type TrackErrors = (errors: string[]) => void
+
+/*
+  This component exists only to capture formik's renderProps and track form
+  submission events. It essentially says:
+  - IF the form has been submitted at least once
+  - AND the form is not submitting at this moment
+  - AND the form is invalid
+  - AND (in useEffect dependencies array) the form submitting state just
+    changed (because it is false, it must have been true above)
+  - THEN run the callback prop.
+  Background:
+    https://github.com/jaredpalmer/formik/issues/1484#issuecomment-490558973
+ */
+const OnSubmitValidationError: React.FC<{
+  cb: TrackErrors
+  formikProps: FormikProps<FormValues>
+}> = props => {
+  const { cb, formikProps } = props
+
+  const effect = () => {
+    if (
+      formikProps.submitCount > 0 &&
+      !formikProps.isSubmitting &&
+      !formikProps.isValid
+    ) {
+      cb(Object.values(formikProps.errors))
+      formikProps.setSubmitting(false)
+    }
+  }
+  React.useEffect(effect, [formikProps.submitCount, formikProps.isSubmitting])
```

Comment:
```suggestion
  React.useEffect(effect, [formikProps.submitCount, formikProps.isSubmitting, formikProps.isValid, formikProps.errors])
```

You need to track all props that are used within the effect.

Reply:
I think this would actually introduce a bug- i know the general way of thinking about it is to track all parts as dependencies, but the purpose of this effect is to run only after failed form submission. Validation runs every time, so by letting `isValid` or `errors` be dependencies we this will change the behavior of the effect. This approach was taken from the formik issue linked in the doc comment and pr comment above & I actually commented on that issue to clarify that my understanding was right (still waiting to hear back).